### PR TITLE
Add table descriptor and schema registry usage for Kafka connector

### PR DESCRIPTION
**Pull request by mosabua**
Replaces the stale progress on https://github.com/trinodb/trino/pull/6791 

It would be great to get this merged before 353 @kokosing

### CODE REVIEW

**Comment by yuhuidavidzhang** [comment]
Update Docs

**Review by kokosing — APPROVED** [review]

File: docs/src/main/sphinx/connector/kafka.rst [hunk 1]
```diff
@@ -61,30 +61,17 @@ The following configuration properties are available:
...
 ========================================================== ==============================================================================
-``kafka.table-names``                                      List of all tables provided by the catalog
 ``kafka.default-schema``                                   Default schema name for tables
 ``kafka.nodes``                                            List of nodes in the Kafka cluster
 ``kafka.buffer-size``                                      Kafka read buffer size
-``kafka.table-description-dir``                            Directory containing topic description files
 ``kafka.hide-internal-columns``                            Controls whether internal columns are part of the table schema or not
 ``kafka.messages-per-split``                               Number of messages that are processed by each Trino split, defaults to 100000
 ``kafka.timestamp-upper-bound-force-push-down-enabled``    Controls if upper bound timestamp push down is enabled for topics using ``CreateTime`` mode
 ========================================================== ==============================================================================
```
Comment (elonazoulay) [inline docs/src/main/sphinx/connector/kafka.rst:70]: Can you include the kafka.table-description-supplier config?
```
``kafka.table-description-supplier``                       The table description supplier to use
```

Comment (mosabua) [reply docs/src/main/sphinx/connector/kafka.rst:70]: No .. it is documented separately and the sentence after the table links to all that config

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -172,8 +151,58 @@ Column name             Type                            Description
...
+.. _kafka-table-schema-registry:
+
+Table schema and schema registry usage
+--------------------------------------
+
+The table schema for the messages can be supplied to the connector with a
+configuration file or a schema registry. It also provides a mechanism for the
+connector to discover tables.
+
+You have to configure the supplier with the ``kafka.table-description-supplier``
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:163]: see here @elonazoulay

Comment (elonazoulay) [reply docs/src/main/sphinx/connector/kafka.rst:163]: LGTM! As long as users can easily see that they have to set that config.

**Comment by mosabua** [comment]
@kokosing or anybody else among  @trinodb/maintainers  ready to merge?

**Comment by kokosing** [comment]
CI hit https://github.com/trinodb/trino/issues/5758

**Comment by kokosing** [comment]
> @kokosing or anybody else among @trinodb/maintainers ready to merge?

I am going to merge it. In case there are more comments we can always apply them as separate pull request.

**Pull request merged**